lib/dateTime.js

```diff
@@ -228,6 +229,9 @@ export function getDateTimeDescriptor() {
 	if (language === 'zh-tw') {
 		hour24 = false;
 	}
+	if (settings.overrides.date && settings.overrides.date.hour24 !== undefined) {
+		hour24 = settings.overrides.date.hour24;
+	}
```

Comment:
This was happening down below when it merges everything together, but the code between here and there actually uses the value, so we need to get the correct potentially overridden value here.